From c25cd73c69b5fa417e6897441f0f195241e9b4f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Wed, 17 Sep 2025 00:22:32 +0200
Subject: [PATCH 074/133] test: skip `alternative constraints` behavior test on
 LoongArch
X-Developer-Signature: v=1; a=openpgp-sha256; l=959; i=xtex@astrafall.org;
 h=from:subject; bh=iWjrKAcJe8OUyc4Ets1p18mPYkABBmMRRg/0JlW4jQg=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78lgNKiwkSuTb7/b+fJwT+uXQs1vfeAJDxMKSW7n4T
 v2+3fajYyILgxgXg6WYIkuRYYM3q046v+iyclmYOaxMIEOkRRoYGBgYWBj4chPzSo10jPRMtQ31
 DA11jHSMGLg4BWCqG78yMvzNeyTNfjJZ83mUSh6DzpqN/a2rZdYE9l87sXj5Yk61GxsY/godYj+
 zovlowbq/q+e48jjzcvk9zH/sVR5VMPGl0Jt2dUYA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

https://github.com/llvm/llvm-project/issues/159200
(cherry picked from commit f90548e74063cd1d6ac6885ee9f6521359cde8cf)
---
 test/behavior/asm.zig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/behavior/asm.zig b/test/behavior/asm.zig
index dae2f579c0ac..ebfd5bb235ab 100644
--- a/test/behavior/asm.zig
+++ b/test/behavior/asm.zig
@@ -67,6 +67,7 @@ test "alternative constraints" {
     if (builtin.zig_backend == .stage2_sparc64) return error.SkipZigTest; // TODO
     if (builtin.zig_backend == .stage2_spirv) return error.SkipZigTest;
     if (builtin.zig_backend == .stage2_riscv64) return error.SkipZigTest;
+    if (builtin.zig_backend == .stage2_llvm and builtin.cpu.arch.isLoongArch()) return error.SkipZigTest; // https://github.com/llvm/llvm-project/issues/159200
 
     if (builtin.zig_backend == .stage2_c and builtin.os.tag == .windows) return error.SkipZigTest; // MSVC doesn't support inline assembly
 
-- 
2.52.0

